### ISSUE TITLE
DropdownMenu items semantics break if the widget is wrapped inside a scrollable view

### ISSUE THREAD

**Issue by lore-co** [flutter/flutter#175184]
### Steps to reproduce

- Create a page with a DropdownMenu
- Wrap the DropdownMenu with a SingleChildScrollView

### Expected results

The semantics for each item should be read when the menu is expanded

### Actual results

As title says, if the DropdownMenu widget is wrapped inside a scrollable view (ListView or SingleChildScrollView), the semantics for its items will be completely ignored.

### Code sample

<details open><summary>Code sample</summary>

```dart
  @override
  Widget build(BuildContext context) {
    final items = ['Item 1', 'Item 2', 'Item 3'];

    return Scaffold(
      body: Center(
        child: SingleChildScrollView(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.center,
            mainAxisSize: MainAxisSize.min,
            children: [
              DropdownButton<String>(
                items: items
                    .map(
                      (e) => DropdownMenuItem(
                        value: e,
                        child: Text(e),
                      ),
                    )
                    .toList(),

                onChanged: (v) {},
              ),
              SizedBox(height: 16),
              DropdownMenu<String>(
                showTrailingIcon: false,
                dropdownMenuEntries: items
                    .map(
                      (e) => DropdownMenuEntry(
                        value: e,
                        label: e,
                      ),
                    )
                    .toList(),
              ),
            ],
          ),
        ),
      ),
    );
  }
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.35.1, on macOS 15.5 24F74 darwin-arm64, locale en-IT)
[✓] Android toolchain - develop for Android devices (Android SDK version 36.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2025.1)
[✓] IntelliJ IDEA Community Edition (version 2023.2.1)
[✓] VS Code (version 1.99.0)
[✓] Connected device (2 available)
[✓] Network resources

```

</details>

**Comment by darshankawar** [comment]
@lore-co Thanks for the report. I verified with talkback ON and below is my observation:

1. On latest stable: The first dropdownmenu is expanded properly and items are read, but when tapping on second dropdownmenu, no action takes place. Is this the issue you are reporting ?
2. On latest master: I am hitting https://github.com/flutter/flutter/issues/171548.

**Comment by lore-co** [comment]
> 2\. [Google3 Bug][A11Y]: MenuAnchor should not allow focus of background items while open #171548

Exactly @darshankawar , DropdownMenu options basically won't be read by talkback if inside a view the user can scroll.
I think it's a little different from the issue you reported since the problem is not about the items behind it, but the items **inside** the menu being totally ignored. 
Also the above issue does not mention the item being inside a scrollview at all

**Comment by lore-co** [comment]
I will add that I tried building my own dropdown using OverlayPortal, but it seems that putting an overlay inside a scrollview will always breaks the semantics of the widget inside the overlayChildBuilder

**Comment by darshankawar** [comment]
Thanks for the update. I was able to replicate the reported behavior on latest sdk versions. 

1. Without wrapping `SingleChildScrollView`, both `DropDownMenu` instances are read properly upon expanding.
2. With wrapping `SingleChildScrollView`, we see below:

- First `DropDownMenu` is expanded and items are read one by one properly.
- Second `DropDownMenu` is expanded but items are not read.

I tried with using below but received same reported behavior:

```
WidgetsFlutterBinding.ensureInitialized();
SemanticsBinding.instance.ensureSemantics();

```
<img width="272" height="219" alt="Image" src="https://github.com/user-attachments/assets/00616313-cf4f-4ccf-a466-59e99257bd56" />
<img width="267" height="260" alt="Image" src="https://github.com/user-attachments/assets/f05f8f83-68dc-499d-a8f1-66c16743ede6" />

```
stable : 3.35.3
master : 3.37.0-1.0.pre-92
```

**Comment by chunhtai** [comment]
cc @QuncCccccc , since you worked on dropdownmenu

**Comment by chunhtai** [comment]
I can't reproduce this, for me it works correctly in latest master

https://github.com/user-attachments/assets/74f081ed-4e44-4492-9f5f-7580d9e15b05

**Comment by chunhtai** [comment]
This is my repro
<details>

```dart
import 'package:flutter/material.dart';
import 'package:flutter_localizations/flutter_localizations.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(useMaterial3: false),
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    final items = ['Item 1', 'Item 2', 'Item 3'];

    return Scaffold(
      body: Center(
        child: SingleChildScrollView(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.center,
            mainAxisSize: MainAxisSize.min,
            children: [
              DropdownButton<String>(
                items:
                    items
                        .map((e) => DropdownMenuItem(value: e, child: Text(e)))
                        .toList(),

                onChanged: (v) {},
              ),
              SizedBox(height: 16),
              DropdownMenu<String>(
                showTrailingIcon: false,
                dropdownMenuEntries:
                    items
                        .map((e) => DropdownMenuEntry(value: e, label: e))
                        .toList(),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

```
</details>

**Comment by lore-co** [comment]
> This is my repro

I've tried your repro.

On 3.35.5 (stable) the issue is still present.
Same on master:

[✓] Flutter (Channel master, 3.37.0-1.0.pre-382, on macOS 26.0 25A354 darwin-arm64, locale en-IT)
[✓] Android toolchain - develop for Android devices (Android SDK version 36.1.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.4)
[✓] Chrome - develop for the web
[✓] Connected device (2 available)
[✓] Network resources

• No issues found!

https://github.com/user-attachments/assets/5963bbef-b5e6-4d3f-b848-e7a1f76fb0fb

**Comment by chunhtai** [comment]
Oh is the issue that the item can't be click to focus? I tried swiping left and right, it is able to focus the item.

**Comment by chunhtai** [comment]
This should be fixed by https://github.com/flutter/flutter/pull/173005, cc @QuncCccccc

**Comment by lore-co** [comment]
> Oh is the issue that the item can't be click to focus? I tried swiping left and right, it is able to focus the item.

Ok it looks like swiping works.

As a user though I would expect:

- to be able to tap on the items
- to be able to swipe vertically since the items are displayed as such

Glad to hear a PR is open on the matter

**Comment by QuncCccccc** [comment]
> This should be fixed by https://github.com/flutter/flutter/pull/173005,

Verified that this PR will fix the problem. Both swiping and tapping on the items work.

https://github.com/user-attachments/assets/96d19a83-c7b7-4080-b0e6-c72833358742
